generate.go

```diff
@@ -65,9 +74,14 @@ func Generate(targetDir string, types []string, pkgName string, newWriter func()
 
 		var importPackages []string
 		if pkgName == "" {
-			pkgName = walker.Pkg.Name
+			pkgName = walkerPkgName
```

Comment:
I think it will produce package name inconsistency if we generate factory in the same directory...!